lib/Doctrine/CouchDB/Utils/BulkUpdater.php

```diff
@@ -83,6 +84,22 @@ public function execute()
         return $this->httpClient->request('POST', $this->getPath(), json_encode($this->data), false, $this->requestHeaders);
     }
 
+    public function executeByLimit($limit = 100)
+    {
+        // Do multiple POST requests if the number of docs is higher than $limit.
+        $result = [];
+        foreach (array_chunk($this->data['docs'], $limit) as $data) {
+            $full_data = $this->data;
+            $full_data['docs'] = $data;
+            $response = $this->httpClient->request('POST', $this->getPath(), json_encode($full_data), false, $this->requestHeaders);
+            if ($response->status != 201) {
+                throw HTTPException::fromResponse($response, $this->getPath());
+            }
+            $result += $response->body;
```

Comment:
This could yield a lot of results at times (the regular one doesn't do it) but do you think it makes any sense to try and implement an iterator of some kind here? That just makes the new request when it needs more results.